### PR TITLE
Dp24 full test

### PR DESCRIPTION
**Pull request by DLBPointon**
Adding Full test for TinyData
Docker testing also added

Adding Sanger specific modified pretext, which changes the memory configurations of the C files ( changes effected by @yumisims ).

#184

### CODE REVIEW

**Comment by DLBPointon** [comment]
Error with PretextSnapshot lacking avx files to deal with CPU architecture.

**Review by weaglesBio — COMMENTED** [review]
Looks okay, will await Yumi & Guoying feedback.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -45,22 +45,27 @@ jobs:
           mkdir -p $NXF_SINGULARITY_CACHEDIR
           mkdir -p $NXF_SINGULARITY_LIBRARYDIR
 
-      - name: Download test data
+      - name: Download Tiny (RAPID) test data
```
Comment (gq1) [inline .github/workflows/ci.yml:48]: It seems this test data being used by rapid test and non-rapid test, why you add a RAPID word here?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -45,22 +45,27 @@ jobs:
...
-      #    nextflow run ${GITHUB_WORKSPACE} -entry RAPID -profile test_github,docker --outdir ./results-rapid
+      # run: |
+      #  nextflow run ${GITHUB_WORKSPACE} -entry RAPID -profile test_github,docker --outdir ./Dock-Rapid
 
-      - name: Singularity - Run RAPID pipeline with test data
+      - name: Singularity - Run FULL pipeline with test data
         # Remember that you can parallelise this by using strategy.matrix
         run: |
-          nextflow run ${GITHUB_WORKSPACE} -entry RAPID -profile test_github,singularity --outdir ./results-rapid
+          nextflow run ${GITHUB_WORKSPACE} -profile test_github,singularity --outdir ./Sing-Full
```
Comment (gq1) [inline .github/workflows/ci.yml:66]: Does this non-rapid test include the steps for rapid test? if so, the rapid test may be not necessary.

File: conf/test_github_full.config [hunk 3]
```diff
@@ -0,0 +1,29 @@
+/*
```
Comment (gq1) [inline conf/test_github_full.config:1]: I can't see this profile being used in any Github workflow

**Comment by gq1** [comment]
If we make the PR only when the branch is nearly ready and reduce the number of pushes, the test workflow will run less.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (yumisims) [reply .github/workflows/ci.yml:66]: agree.

File: conf/test_github_full.config (hunk 3 above)
Comment (yumisims) [reply conf/test_github_full.config:1]: should change back to the non paid server

File: .github/workflows/ci.yml (hunk 2 above)
Comment (DLBPointon) [reply .github/workflows/ci.yml:66]: Yes, rapid is a subset of full. I'll remove rapid from ci.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (DLBPointon) [reply .github/workflows/ci.yml:48]: This was whilst I was putting together a bigger test data set which we no longer need. I'll remove.

File: conf/test_github_full.config (hunk 3 above)
Comment (DLBPointon) [reply conf/test_github_full.config:1]: Changed the CI to use the full profile.

The local testing yamls are so we all use the same yaml for test runs, we haven't been using these as part of any profile although I could set them up as such.

**Comment by DLBPointon** [comment]
> If we make the PR only when the branch is nearly ready and reduce the number of pushes, the test workflow will run less.

That's fair, I'll do this in the future.

File: conf/test_github_full.config (hunk 3 above)
Comment (DLBPointon) [reply conf/test_github_full.config:1]: The BGA.yaml is specific to the gitpod that is included in the repo.

**Review by gq1 — APPROVED** [review]
Looks good for me.

**Pull request merged**